src/reusable-blocks/class-manage-batch.php

```diff
@@ -4,10 +4,30 @@
 
 class ManageBatch extends Manage {
 
+	/** @var Notice $notice */
+	private $notice;
+
+	public function __construct(
+		Blocks $blocks,
+		Translation $translation,
+		Notice $notice
+	) {
+		parent::__construct( $blocks, $translation );
+		$this->notice = $notice;
+	}
+
 	public function addBlocks( \WPML_TM_Translation_Batch $batch ) {
 		$blocks = $this->getBlocksFromPostElements( \collect( $batch->get_elements() ) );
 
-		$this->getBlockElementsToAdd( $blocks )->each(
+		$blocks_to_add = $this->getBlockElementsToAdd( $blocks );
+
+		if ( $blocks_to_add->count() < 1 ) {
+			return $batch;
+		}
+
+		add_action( 'wpml_added_translation_jobs', [ $this, 'notifyExtraJobsToTranslator' ] );
```

Comment:
@strategio Adding an action that then calls another function in this class is a type of spaghetti code. Don't we know already in this function?

Otherwise I think you move the hook to the `Notice` class and responds to the event without involving this class.

Reply:
@brucepearson, we have 2 ways to add the blocks:
1. Automatically create jobs when we click on the "+" icon (we add the blocks to the batch).
2. Automatically add block elements to the basket (then the translation manager can decide to keep it or not).

We want this notice to show up only when we click on the "+" icon. That's why we have the method `notifyExtraJobsToTranslator` in this class.

I could add the hook in the `AdminIntegration` class and set a private property (i.e. `$has_block_jobs`) in `ManageBatch` to know if we need this notice or not.

In the end, the logic is almost the same: We have to hook to `wpml_added_translation_jobs` only when we have blocks added in this `ManageBatch::addBlocks`.

Reply:
Finally, I found a better way to do this @brucepearson. I think I wanted to overoptimize the code to add the hook only when we have blocks in a direct job. Now it's clearer at least.